core/src/main/java/greencity/converters/UserArgumentResolver.java

```diff
@@ -38,6 +41,10 @@ public boolean supportsParameter(MethodParameter parameter) {
     public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer,
                                   NativeWebRequest webRequest, WebDataBinderFactory binderFactory) {
         Principal principal = webRequest.getUserPrincipal();
+        if ((parameter.getParameterType().equals(UserVO.class))) {
+            return principal != null ? modelMapper.map(userService.findByEmail(principal.getName()),
+                UserVO.class) : null;
+        }
```

Comment:
Also you can move this logic to private method, smth like:
return principal != null ? privateMethodName() : null;

Reply:
With service movement it will be fixed.